### PR TITLE
Allow multiple extra_packages in Dataflow

### PR DESCRIPTION
**Pull request by mik-laj**
Fix for: https://issues.apache.org/jira/browse/AIRFLOW-2153
---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 1]
```diff
@@ -681,14 +681,25 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
         ]
-        if variables is not None:
-            for attr, value in variables.items():
-                if attr == 'labels':
-                    command += label_formatter(value)
-                elif value is None or value.__len__() < 1:
+        if variables is None:
+            return command
+
+        # The logic of this method should be compatible with Apachea Beam:
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
        # The logic of this method should be compatible with Apache Beam:
```

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 2]
```diff
@@ -681,14 +681,25 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+        # https://github.com/apache/beam/blob/b56740f0e8cd80c2873412847d0b336837429fb9/sdks/python/
+        # apache_beam/options/pipeline_options.py#L230-L251
+        for attr, value in variables.items():
+            if attr == 'labels':
+                command += label_formatter(value)
+            elif value is None:
+                command.append("--" + attr)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
                command.append(f"--{attr}")
```
WDYT?

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 3]
```diff
@@ -681,14 +681,25 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+            elif isinstance(value, bool):
+                if value:  # pylint: disable=too-many-nested-blocks
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
            elif isinstance(value, bool) and value:
```

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 4]
```diff
@@ -681,14 +681,25 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
                     command.append("--" + attr)
-                else:
-                    command.append("--" + attr + "=" + value)
+            elif isinstance(value, list):
+                for v in value:  # pylint: disable=too-many-nested-blocks
+                    command.append("--" + attr + "=" + v)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
                command.extend([f"--{attr}={v}" for v in value])
```

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 5]
```diff
@@ -681,14 +681,25 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+            else:
+                command.append("--" + attr + "=" + value)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
                command.append(f"--{attr}={value}")
```

**Review by turbaszek — APPROVED** [review]
Small suggestions otherwise LGTM

File: airflow/providers/google/cloud/hooks/dataflow.py (hunk 3 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/dataflow.py]: Here `value is True` will also work

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 6]
```diff
@@ -681,14 +681,24 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+        # https://github.com/apache/beam/blob/b56740f0e8cd80c2873412847d0b336837429fb9/sdks/python/
+        # apache_beam/options/pipeline_options.py#L230-L251
+        for attr, value in variables.items():
+            if attr == 'labels':
+                command += label_formatter(value)
+            elif value is None:
+                command.append(f"--{attr}")
+            elif isinstance(value, bool):
+                if value:  # pylint: disable=too-many-nested-blocks
                     command.append("--" + attr)
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
            elif isinstance(value, bool) and value
                command.append("--" + attr)
```

File: airflow/providers/google/cloud/hooks/dataflow.py (hunk 3 above)
Comment (mik-laj) [reply airflow/providers/google/cloud/hooks/dataflow.py]: Thanks.  Did you know that you can already add suggestions on multiple lines?
https://github.blog/changelog/2020-02-26-multi-line-code-suggestions-beta/
Demo: 
https://github.com/apache/airflow/pull/8394#discussion_r410908722
Thanks to this, you can quickly edit files from Web UI.

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 7]
```diff
@@ -681,14 +681,23 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+        # The logic of this method should be compatible with Apache Beam:
+        # https://github.com/apache/beam/blob/b56740f0e8cd80c2873412847d0b336837429fb9/sdks/python/
+        # apache_beam/options/pipeline_options.py#L230-L251
+        for attr, value in variables.items():
+            if attr == 'labels':
+                command += label_formatter(value)
+            elif value is None:
+                command.append(f"--{attr}")
+            elif isinstance(value, bool) and value
+                command.append("--" + attr)
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
                command.append(f"--{attr}")
```

File: airflow/providers/google/cloud/hooks/dataflow.py (hunk 3 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/dataflow.py]: Yeah I know

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 8]
```diff
@@ -681,14 +681,23 @@ def _build_cmd(variables: Dict, label_formatter: Callable, project_id: str) -> L
...
+
+        # The logic of this method should be compatible with Apache Beam:
+        # https://github.com/apache/beam/blob/b56740f0e8cd80c2873412847d0b336837429fb9/sdks/python/
+        # apache_beam/options/pipeline_options.py#L230-L251
+        for attr, value in variables.items():
+            if attr == 'labels':
+                command += label_formatter(value)
+            elif value is None:
+                command.append(f"--{attr}")
+            elif isinstance(value, bool) and value
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/dataflow.py]:
```suggestion
            elif isinstance(value, bool) and value:
```

**Pull request merged**